lib/stat_tracker.rb

```diff
@@ -181,18 +191,48 @@ def coach_stats
     end
     coach_percentages
   end
+  
+  def find_season(season_id)
+    @seasons.find do |season|
+      season.season == season_id
+    end
+  end
 
-  def winningest_coach
-    best_coach_stats = coach_stats.max_by {|coach, percentage| percentage}
+  def winningest_coach(season)
+    best_coach_stats = coach_stats(season).max_by {|coach, percentage| percentage}
     best_coach_stats[0]
   end
 
-  def worst_coach
-    worst_coach_stats = coach_stats.min_by {|coach, percentage| percentage}
+  def worst_coach(season)
+    worst_coach_stats = coach_stats(season).min_by {|coach, percentage| percentage}
     worst_coach_stats[0]
   end
 
-  def most_accurate_team
-    
+  def most_accurate_team(season)
+    id_hash = find_season(season).game_teams.group_by { |game_team_object| game_team_object.team_id }
+    team_percentages = {}
+    id_hash.each do |team_id, games_array|
+      team_accuracy = games_array.map { |game_team_object| game_team_object.accuracy }
+      accurate_percentage = team_accuracy.sum.to_f / team_accuracy.length
+      team_percentages[team_id] = accurate_percentage
+    end
+    top_team_array = team_percentages.max_by {|team_id, percentage| percentage }
+    top_team_id = top_team_array[0]
+    top_team = @teams.find {|team| team.id == top_team_id }
+    top_team.team_name
+  end
+
+  def least_accurate_team(season)
+    id_hash = find_season(season).game_teams.group_by { |game_team_object| game_team_object.team_id }
+    team_percentages = {}
+    id_hash.each do |team_id, games_array|
+      team_accuracy = games_array.map { |game_team_object| game_team_object.accuracy }
+      accurate_percentage = team_accuracy.sum.to_f / team_accuracy.length
+      team_percentages[team_id] = accurate_percentage
+    end
+    top_team_array = team_percentages.min_by {|team_id, percentage| percentage }
```

Comment:
212 through 218, and 226 through 232 are identical. We can put those in a helper method the same way we made a `coach_stats` helper and a `scoring_helper_method`